src/user_input_compiler.rs

```diff
@@ -245,13 +241,27 @@ fn collect_real(s: &mut Chars) -> Result<String, Input> {
   Ok( i1 )
 }
 
+fn match_full_integer(s :&mut Chars) -> Result<i64, Input> {
+  let _ = consume_whitespaces(s);
+  let sign = if let Some(c) = s.clone().next() { if c == '-' { -1 } else { 1 } } else { 1 };
+  
+  if sign < 0 {
+    s.next();
+  }
+  
+  
+  
+  let st :String = try!(collect_integer(s));
+  match st.parse::<i64>() {
+    Ok(val)  => Ok(sign * val),
+    Err(msg) => Err( InvalidSyntax(format!("Invalid Integer: {:?}", msg)) ),
+  }
+}
```

Comment:
For all this parsing stuff, you might be better off with a parsing library like https://github.com/Geal/nom. I think it already provides combinators for things like "match an integer".